### PR TITLE
Ean/make notifier hosted aware

### PR DESCRIPTION
**Pull request by ean**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

@mpolden

### CODE REVIEW

File: controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/notify/NotifyDispatcher.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import com.yahoo.vespa.hosted.controller.tenant.TenantContacts;
+
+import java.util.Collection;
+
+/**
+ * NotifyDispatcher defines the notify API
+ *
+ * @author enygaard
+ */
+public interface NotifyDispatcher {
```
Comment (mpolden) [inline controller-api/src/main/java/com/yahoo/vespa/hosted/controller/api/integration/notify/NotifyDispatcher.java:13]: Why is this needed in addition to the existing `Mailer` interface?

**Comment by ean** [comment]
We need the extra indirection to separate fetching of contact data from creating the mail content.
CuratorDB lives in vespa-engine/vespa, while hosted information like the dashboard url lives in the close repository. 

And to avoid exposing all of the Hosted System information in `vespa` the indirection here is needed.

It will also give us an extendable way to add other destinations in the future (Slack, IRC, ...) now that Notifier is no longer responsible for dispatching a notification to the destination.
